File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/util/PercentUtils.java
/* Copyright Airship and Contributors */

package com.urbanairship.android.layout.util;

import java.util.regex.Pattern;

import androidx.annotation.NonNull;

public final class PercentUtils {
    private static final Pattern PATTERN_NON_NUMERIC = Pattern.compile("[^\\d.]");
    private static final Pattern PATTERN_PERCENT_W_SYMBOL = Pattern.compile("^\\d{1,3}%$");

    private PercentUtils() {}

    public static boolean isPercent(@NonNull String string) {
        return PATTERN_PERCENT_W_SYMBOL.matcher(string).matches();
    }

    public static float parse(@NonNull String percentString) {
        String number = digits(percentString);
        return Float.parseFloat(number) / 100f;
    }

    @NonNull
    public static String digits(@NonNull String number) {
        return PATTERN_NON_NUMERIC.matcher(number).replaceAll("");
    }
}

File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/util/PercentUtils.kt
/* Copyright Airship and Contributors */
package com.urbanairship.android.layout.util

import java.util.regex.Pattern

public object PercentUtils {

    private val PATTERN_NON_NUMERIC: Pattern = Pattern.compile("[^\\d.]")
    private val PATTERN_PERCENT_W_SYMBOL: Pattern = Pattern.compile("^\\d{1,3}%$")

    public fun isPercent(string: String): Boolean {
        return PATTERN_PERCENT_W_SYMBOL.matcher(string).matches()
    }

    public fun parse(percentString: String): Float {
        val number = digits(percentString)
        return number.toFloat() / 100f
    }

    public fun digits(number: String): String {
        return PATTERN_NON_NUMERIC.matcher(number).replaceAll("")
    }
}
